src/components/MemoryCard.jsx

```diff
@@ -1,5 +1,10 @@
-function MemoryCard() {
-  return <p>Emojis</p>;
+function MemoryCard({ handleClick, data }) {
+  const emojiEl = data.map((emojiElement, index) => (
+    <li key={index}>
+      <button onClick={handleClick}>{emojiElement.emoji}</button>
+    </li>
+  ));
+  return <ul className="card-container">{emojiEl}</ul>;
```

Comment:
The is no need to place the `map` to a constant. This can be rendered inside the JSX
```suggestion
function MemoryCard({ handleClick, data }) {
  return <ul className="card-container">
    {data.map((emojiElement, index) => (
      <li key={index}>
        <button onClick={handleClick}>{emojiElement.emoji}</button>
      </li>
    ))}
  </ul>;
```

Reply:
Edge cases are not included. What if the data is null or empty array []? The UI should handle those states.

Reply:
good recommendation thank you.